apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Auth.OAuth2;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Security.Cryptography;
+using Xunit;
+
+#if NETCOREAPP1_0
+using RsaKey = System.Security.Cryptography.RSA;
+#else
+using RsaKey = System.Security.Cryptography.RSACryptoServiceProvider;
+#endif
+
+namespace Google.Cloud.Storage.V1.Tests
+{
+    public class UrlSignerTest
+    {
+        [Fact]
+        public void FromServiceAccountPath_Validations()
+        {
+            Assert.Throws<ArgumentNullException>(() => UrlSigner.FromServiceAccountPath(null));
+            Assert.Throws<DirectoryNotFoundException>(() => UrlSigner.FromServiceAccountPath("C:\\does_not_exist\\foo.txt"));
+        }
+
+        [Fact]
+        public void FromServiceAccountData_Validations()
+        {
+            Assert.Throws<ArgumentNullException>(() => UrlSigner.FromServiceAccountData(null));
+        }
+
+        [Fact]
+        public void FromServiceAccountCredential_Validations()
+        {
+            Assert.Throws<ArgumentNullException>(() => UrlSigner.FromServiceAccountCredential(null));
+        }
+
+        [Fact]
+        public void Sign_Validations()
+        {
+            var signer = UrlSigner.FromServiceAccountCredential(new ServiceAccountCredential(new ServiceAccountCredential.Initializer("test")
```

Comment:
Possibly extract this to a separate method, given that we've got it in a few places?